src/components/PostsList/PostsList.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../types';
+
+type Props = {
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
+};
+
+export const PostsList: React.FC<Props> = ({
+                                             userSelectedId,
+                                             selectPost,
+                                             post,
+                                           }) => {
+  const [currentPostList, setPostList] = useState<Post[]>([]);
+
+  const allPosts = async () => {
+    const result = await getAllPosts();
+
+    setPostList(result);
+  };
+
+  const findposts = async () => {
+    const result = await getPosts(userSelectedId);
+
+    setPostList(result);
+  };
+
+  useEffect(() => {
+    findposts();
+  }, [userSelectedId]);
```

Comment:
the method `findposts` is missing from the `useEffect` dependency array, or we add it to the array, or we move the function to inside the `useEffect` hook, since it is only used inside the hook.